bind/gencffi.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2017 The go-python Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package bind
+
+import (
+	"go/token"
+)
+
+const (
+/*
+FIXME(corona10): ffibuilder.cdef should be written this way.
+ffibuilder.cdef("""
+	//header exported from 'go tool cgo'
+	#include "%[3]s.h"
+	""")
+
+* discuss: https://github.com/go-python/gopy/pull/93#discussion_r119652220
+*/
+	builderPreamble = `import os
+
+from sys import argv
+from cffi import FFI
+
+ffibuilder = FFI()
+ffibuilder.set_source(
+    '%[1]s',
+    None,
+    extra_objects=["_%[1]s.so"],
+)
+
+ffibuilder.cdef("""
+typedef signed char GoInt8;
+typedef unsigned char GoUint8;
+typedef short GoInt16;
+typedef unsigned short GoUint16;
+typedef int GoInt32;
+typedef unsigned int GoUint32;
+typedef long long GoInt64;
+typedef size_t GoUintptr;
+typedef unsigned long long GoUint64;
+typedef GoInt64 GoInt;
+typedef GoUint64 GoUint;
+typedef float GoFloat32;
+typedef double GoFloat64;
+typedef struct { const char *p; GoInt n; } GoString;
+typedef void *GoMap;
+typedef void *GoChan;
+typedef struct { void *t; void *v; } GoInterface;
+typedef struct { void *data; GoInt len; GoInt cap; } GoSlice;
+
+extern GoString _cgopy_GoString(char* p0);
+extern char* _cgopy_CString(GoString p0);
+extern GoUint8 _cgopy_ErrorIsNil(GoInterface p0);
+extern char* _cgopy_ErrorString(GoInterface p0);
+extern void cgopy_incref(void* p0);
+extern void cgopy_decref(void* p0);
+
+extern void cgo_pkg_%[1]s_init();
+
+`
+	builderPreambleEnd = `""")
+
+if __name__ == "__main__":
+    ffibuilder.compile()
+    # Remove itself after compile.
+    os.remove(argv[0])
+`
+
+	cffiPreamble = `
+__doc__="""%[1]s"""
+import os
+
+here = os.path.dirname(os.path.abspath(__file__))
+lib = ffi.dlopen(os.path.join(here, "_%[2]s.so"))
+
+# python <--> cffi helpers.
+def cffi_cnv_py2c_string(o):
```

Comment:
it might be better to not export all these `cffi_cnv_xxx` function.
or at least, clearly say they shouldn't be used by prepending a `_` in front of the `cffi_cnv_xxx`.

ie:
`def _cffi_cnv_py2c_string(o):`